src/object.js

```diff
@@ -102,27 +132,302 @@ Sk.builtin.object.prototype.tp$setattr = Sk.builtin.object.prototype.GenericSetA
+ * @type {string}
+ */
+Sk.builtin.object.prototype.tp$name = "object";
+
+/**
+ * The type object of this class.
+ * @type {Sk.builtin.type}
+ */
 Sk.builtin.object.prototype.ob$type = Sk.builtin.type.makeIntoTypeObj("object", Sk.builtin.object);
 
+/** Default implementations of dunder methods found in all Python objects */
+
+/**
+ * Python wrapper for `__repr__` method.
+ * @name  __repr__
+ * @memberOf Sk.builtin.object.prototype
+ * @instance
+ */
+Sk.builtin.object.prototype["__repr__"] = function (self) {
+    Sk.builtin.pyCheckArgs("__repr__", arguments, 0, 0, false, true);
+
+    return self["$r"]();
+};
+
+/**
+ * Python wrapper for `__str__` method.
+ * @name  __str__
+ * @memberOf Sk.builtin.object.prototype
+ * @instance
+ */
+Sk.builtin.object.prototype["__str__"] = function (self) {
+    Sk.builtin.pyCheckArgs("__str__", arguments, 0, 0, false, true);
+
+    return self["$r"]();
+};
+
+/**
+ * Python wrapper for `__hash__` method.
+ * @name  __hash__
+ * @memberOf Sk.builtin.object.prototype
+ * @instance
+ */
+Sk.builtin.object.prototype["__hash__"] = function (self) {
+    Sk.builtin.pyCheckArgs("__hash__", arguments, 0, 0, false, true);
+
+    return self.tp$hash();
+};
+
+/**
+ * Python wrapper for `__eq__` method.
+ * @name  __eq__
+ * @memberOf Sk.builtin.object.prototype
+ * @instance
+ */
+Sk.builtin.object.prototype["__eq__"] = function (self, other) {
+    Sk.builtin.pyCheckArgs("__eq__", arguments, 1, 1, false, true);
+
+    return self.ob$eq(other);
+};
+
+/**
+ * Python wrapper for `__ne__` method.
+ * @name  __ne__
+ * @memberOf Sk.builtin.object.prototype
+ * @instance
+ */
+Sk.builtin.object.prototype["__ne__"] = function (self, other) {
+    Sk.builtin.pyCheckArgs("__ne__", arguments, 1, 1, false, true);
+
+    return self.ob$ne(other);
+};
+
+/**
+ * Python wrapper for `__lt__` method.
+ * @name  __lt__
+ * @memberOf Sk.builtin.object.prototype
+ * @instance
+ */
+Sk.builtin.object.prototype["__lt__"] = function (self, other) {
+    Sk.builtin.pyCheckArgs("__lt__", arguments, 1, 1, false, true);
+
+    return self.ob$lt(other);
+};
+
+/**
+ * Python wrapper for `__le__` method.
+ * @name  __le__
+ * @memberOf Sk.builtin.object.prototype
+ * @instance
+ */
+Sk.builtin.object.prototype["__le__"] = function (self, other) {
+    Sk.builtin.pyCheckArgs("__le__", arguments, 1, 1, false, true);
+
+    return self.ob$le(other);
+};
+
+/**
+ * Python wrapper for `__gt__` method.
+ * @name  __gt__
+ * @memberOf Sk.builtin.object.prototype
+ * @instance
+ */
+Sk.builtin.object.prototype["__gt__"] = function (self, other) {
+    Sk.builtin.pyCheckArgs("__gt__", arguments, 1, 1, false, true);
+
+    return self.ob$gt(other);
+};
+
+/**
+ * Python wrapper for `__ge__` method.
+ * @name  __ge__
+ * @memberOf Sk.builtin.object.prototype
+ * @instance
+ */
+Sk.builtin.object.prototype["__ge__"] = function (self, other) {
+    Sk.builtin.pyCheckArgs("__ge__", arguments, 1, 1, false, true);
+
+    return self.ob$ge(other);
+};
+
```

Comment:
Hi @mchat 

So I was looking at another problem today and started experimenting a bit.  And I think this is not quite right.  Here's the example:

```
class Foo1(object):
    def __add__(self, other):
        if isinstance(other, Foo2):
            print "OK"

class Foo2(object):
    def __init__(self):
        pass

x = Foo1()
y = Foo2()

# this goes OK
z = x + y

print x.__ge__(y)
```
Now this fails with a wrong argument count.

I tracked it through to line 225 in abstract.js (which has been the same since 2010 I would add) where it says
``vop.call(v,w)``  The problem of course is that call treats ``v`` as ``this`` and then w is undefined, hence the wrong argument count.

But I think the bigger issue is that in iboNameToSlotFunc_ there was an assumption that ``obj['__xxx__')`` would be a user created python function not an internal javascript function.

What do you think the right way to go is?